File: app/models/review.rb
class Review < ActiveRecord::Base
  RATINGS = (1..10).to_a
  belongs_to :user
  belongs_to :course

  validates :post, :rating, :user, :course, presence: true
  validates :rating, inclusion: RATINGS
  validates :user, uniqueness: { scope: :course }
end

File: app/assets/javascripts/models/review.js
App.Review = DS.Model.extend({
  rating: DS.attr('number'),
  post: DS.attr('string'),
  course: DS.belongsTo('course', { async: true }),
  user: DS.belongsTo('user', { async: true }),
  createdAt: DS.attr('date'),

  userFullName: Ember.computed.alias('user.fullName')
});
